src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
```

Comment:
We can optimize this solution by avoiding double enumeration. Create an object with `name` as a key before `children` variable definition
```
const peopleObject = people.reduce((acc, p) => ({...acc, [p.name]: p}),{})
```
and then we can replace `people.some(m => m.name === p.mother)` by `peopleObject[p.mother]`